src/Node.js

```diff
@@ -0,0 +1,133 @@
+  // ### DOM Node Shim
+  //
+  // This represents a superficial form of a DOM node which contains most of
+  // the properties that a regular DOM node would contain, but only the ones
+  // needed by slowparse.
+module.exports = (function(){
+  "use strict";
+
+  function Node(nodeValue, namespaceURI) {
+    var self = this;
+    self.nodeValue = nodeValue || "";
+    self.namespaceURI = namespaceURI || "";
+    self.parentNode = false;
+    self.nextSibling = false;
+    self.isVoid = false;
+    self.childNodes = [];
+    self.attributes = [];
+    Object.defineProperties(self, {
+      "innerHTML": {
+        get: function() {
+          if(self.nodeType !== self.ELEMENT_NODE) {
+            return null;
+          }
+
+          var innerHTML = "";
+
+          self.childNodes.forEach(function(childNode) {
+            innerHTML += childNode.outerHTML || "";
+          });
+
+          return innerHTML;
+        }
+      },
+      "outerHTML": {
+        get: function() {
+          if(self.nodeType === self.ATTRIBUTE_NODE) {
+            return null;
+          }
+          if(self.nodeType === self.COMMENT_NODE) {
+            return "<!--" + self.nodeValue + "-->";
+          }
+          if(self.nodeType !== self.ELEMENT_NODE) {
+            return self.nodeValue;
+          }
+
+          var attributeStr = "";
```

Comment:
use `map` and join.
```
var attributeStr = this.attributes.map(function(..) { return ... }).join(" ");
```
However, later code also suggests `this.attributes` should be an object, not an array, which means we'd do:
```
var attributeStr = Object.keys(this.attributes).map(function(k) { return k + "=" + attributes[k]; }).join(" ");
```